app/src/main/java/edu/uc/hickmadc/customnotifications/ui/main/dialogfragment.kt

```diff
@@ -5,30 +5,58 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
+import android.widget.ImageButton
+import android.widget.TextView
 import androidx.fragment.app.DialogFragment
 import androidx.fragment.app.FragmentContainer
 import edu.uc.hickmadc.customnotifications.R
+import edu.uc.hickmadc.customnotifications.dto.Notification
 import java.io.FileDescriptor
 import java.io.PrintWriter
 import java.util.zip.Inflater
 
 
+class dialogfragment : DialogFragment() {
 
-class dialogfragment: DialogFragment() {
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
-    ): View?{
+    private lateinit var rootView: View
+    private lateinit var notificationTitleText: TextView
+    private lateinit var notificationSubtext: TextView
+    private lateinit var notificationText: TextView
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
         //gets dialog fragment
-        var rootView: View= inflater.inflate(R.layout.notification_dialog,container,false)
+        rootView = inflater.inflate(R.layout.notification_dialog, container, false)
+
+        notificationTitleText = rootView.findViewById(R.id.txtTitle)
+        notificationSubtext = rootView.findViewById(R.id.txtSubtext)
+        notificationText = rootView.findViewById(R.id.notificationText)
+
+        addEventHandlersToDialogBox()
```

Comment:
Here I bind the variables I created above to the ui components. Having this logic here is definitely needed as before you were not storing the values the user entered into the dialog box. I still don't store them for you, but I gave you a head start by retrieving them from the user interface.